### PR TITLE
feat: add Go scaffolding

### PR DESCRIPTION
**Pull request by alexeagle**
---

### Changes are visible to end-users: yes

- Searched for relevant documentation and updated as needed: yes
- Breaking change (forces users to change their own code or config): yes
- Suggested release notes appear below: yes

Now offers to setup rules_go.

### Test plan

- New test cases added

### CODE REVIEW

File: scaffold.yaml [hunk 1]
```diff
@@ -59,29 +60,36 @@ features:
     globs:
       - "*/package.json"
       - "*/pnpm-*.yaml"
-      - "*/.eslintrc*"
+      - "*/eslint.config.mjs*"
```
Comment (jbedard) [inline scaffold.yaml:63]: is this related to go?

**Review by jbedard — APPROVED** [review]

File: scaffold.yaml (hunk 1 above)
Comment (alexeagle) [reply scaffold.yaml:63]: no sorry, it was a drive-by cleanup from the eslint upgrade where I missed this, and caused non-JS projects to get this file.
